### PR TITLE
Boos 185 read single role

### PR DESCRIPTION
**Pull request by anibalgatica**
Read single role, read role list updated to use the API endpoint instead of mock data.

### CODE REVIEW

**Review by joaopcartier — APPROVED** [review]

**Review by joaopcartier — APPROVED** [review]

File: backend/project/api/routes/roles_routes.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ def get_roles():
 
 @bp.route("/roles", methods=["POST"])
 def create_role():
+    print(request)
```
Comment (joaopcartier) [inline backend/project/api/routes/roles_routes.py:20]: I guess this print was created when we were having problems with connecting with the API.
Don't think it's meaningful to the backend code

**Review by joaopcartier — CHANGES_REQUESTED** [review]

File: frontend/src/components/role/RoleDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    ) => {
+        setFields({
+            ...fields,
+            [fieldKey]: value,
+        });
+    }
+
+// const handleSaveRole = async () => {
+//     const response = await dispatch(updateRole({id: role.id, fields}))
+// }
```
Comment (koutsilisk) [inline frontend/src/components/role/RoleDetails.tsx:112]: do we need this commented code ?

File: frontend/src/components/role/RoleDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+            <Dialog 
+                message={`Are you sure that you want to delete the role ${role.name}? This cannot be undone.`}
+                open={showDeleteModal}
+                // options={[
+                //     {action: async () => {
+                //         setShowDeleteModal(false);
+                //         await dispatch(remove(role?.id || -1));
+                //         await dispatch(requestRoles());
+                //     }}
+                // ]}
```
Comment (koutsilisk) [inline frontend/src/components/role/RoleDetails.tsx:133]: same

File: frontend/src/components/role/RoleDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+              </Button>
+              <span
+                onClick={() => setEditing(false)}
+                className="text-center text-gray cursor-pointer"
+              >
+                Cancel
+              </span>
+            </>
+          )}
+        </div> */}
```
Comment (koutsilisk) [inline frontend/src/components/role/RoleDetails.tsx:197]: same

File: frontend/src/components/role/RolesList.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+          loading={rolesData.status === "loading"}
+          customRenderers={{
+          delete: (item: Role) => (
+            <Close
+              onClick={() => {}}
+              // onClick={(e: React.MouseEvent<HTMLButtonElement>) => {
+              //   e.stopPropagation();
+              //   // alert(`Deleting ${role.name}`);
+              //   setDeleteRoleSelected(item);
+              // }}
```
Comment (koutsilisk) [inline frontend/src/components/role/RolesList.tsx:69]: same

File: frontend/src/components/role/roleAPI.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { Role } from "types/role";
+import { request, RequestError } from 'utils/api';
+
+export const fetchRoles = async () => {
+  const response = await request("http://localhost:5000/api/roles");
+  console.log({ response });
+  return response;
+  };
```
Comment (koutsilisk) [inline frontend/src/components/role/roleAPI.ts:8]: add new line at end of file

File: frontend/src/components/role/roleSlice.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+          state.status = "loading";
+        })
+        .addCase(requestRoles.fulfilled, (state, action) => {
+          state.status = "idle";
+          console.log("Roles: ", { action });
+          state.roles = action.payload || [];
+        })
+        // .addCase(createRole.fulfilled, (state, action) => {
+        //   sendNotification("error", )
+        // })
```
Comment (koutsilisk) [inline frontend/src/components/role/roleSlice.ts:55]: same for commented code here

File: frontend/src/components/role/roleSlice.ts [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+    }
+  });
+
+  export default roleSlice.reducer
```
Comment (koutsilisk) [inline frontend/src/components/role/roleSlice.ts:59]: new line

File: frontend/src/components/user/userAPI.ts [hunk 9]
```diff
@@ -4,7 +4,7 @@ import { CreateUserFormValues } from "./CreateUserForm";
 
 // A mock function to mimic making an async request for data
 export const fetchUsers = async (): Promise<User[]> => {
-  const response = await request("http://localhost:8874/api/users");
+  const response = await request("http://localhost:5000/api/users");
```
Comment (koutsilisk) [inline frontend/src/components/user/userAPI.ts:7]: can http://localhost:5000 be a "global" variable and use something like HOST/api/users ? This way you just update the var in one place and can reuse it everywhere

File: frontend/src/types/role.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+export interface Role {
+    id: Number;
+    name: string;
+    description: string;
+    group: string;
+    permissions: string;
+    created_at: string;
+    updated_at: string;
+  }
```
Comment (koutsilisk) [inline frontend/src/types/role.ts:9]: new line

**Review by koutsilisk — CHANGES_REQUESTED** [review]

File: frontend/src/components/role/roleAPI.ts (hunk 6 above)
Comment (anibalgatica) [reply frontend/src/components/role/roleAPI.ts:8]: It doesn't cause any harm, but it's not required in javascript or typescript

File: frontend/src/components/user/userAPI.ts (hunk 9 above)
Comment (anibalgatica) [reply frontend/src/components/user/userAPI.ts:7]: It's actually a good idea, not necessary to do here I think but I'll take a note to do it later.

**Pull request merged**